### ISSUE TITLE
Question 2-0/authentication : Multiple Authentication Types

### ISSUE THREAD

**Issue by mischac** [swagger-api/swagger-codegen#6644]
##### Description
From https://swagger.io/docs/specification/2-0/authentication/#multiple I understand that one can distinguish re **AND** vs **OR** in case there are more than one securities for an operation.

However in  gen model `CodegenOperation` we have 
`public List<CodegenSecurity> authMethods;`.

So it is unclear how a generator can deal with different combinations for multiple securities for an operation.

Version `2.2.3`

From link above:
```
security:    # A OR B
  - A
  - B
security:    # A AND B
  - A
    B
security:    # (A AND B) OR (C AND D)
  - A
    B
  - C
    D
```

**Comment by wing328** [comment]
@mischac yes, you're right. There's no way to tell at the moment and we would welcome contribution to add the support.

For API clients, we've a workaround for the issue at the moment.
